pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -3425,6 +3425,14 @@ def _check_function(
         # Check that all positional arguments are correctly annotated.
         if match.arg_types:
             for key, expected_type in match.arg_types.items():
+                if key > len(node.args.args) - 1:
+                    # The number of arguments is less than expected
+                    self.add_message(
+                        "hass-argument-type",
```

Comment:
Would it make sense to add a dedicated pylint error code for it? Something like `hass-missing-argument`?